test/gtest/mhp/halo.cpp

```diff
@@ -0,0 +1,39 @@
+#include "xhp-tests.hpp"
+#include <dr/mhp/views/sliding.hpp>
+
+template <typename T> class Halo : public testing::Test {};
+
+TYPED_TEST_SUITE(Halo, AllTypes);
+
+TYPED_TEST(Halo, dv_different_halos) {
+  TypeParam dv(6, dr::mhp::distribution().halo(1, 2));
+  std::vector<int> v(6);
+
+  iota(dv, 1);
+  std::iota(v.begin(), v.end(), 1);
+  dv.halo().exchange();
+
+  EXPECT_TRUE(equal(dv, v));
```

Comment:
Test does not check halo. I think you could remove line with `dv.halo().exchange()` and it will still pass. This is test of `iota` currently :)

I don't have any idea about sensible checks to write here which are not covered by `Halo3` suite. If you too, then let's remove this test.